### DISCUSSION TITLE
the problem of the colormap?

### DISCUSSION THREAD

**Discussion by keepbeer** [developmentseed/titiler#977] [Q&A]
i don't know how to use the colormap, is anyone here to tell me how to use it in titiler mosaicjson api?thanks

**Comment by zacdezgeo** [comment]
Hi @kl402401! Did you look at `<your_deployment>/API.html`? It provides docs to the endpoints. If running locally with docker-compose, you should be able to get more details and perhaps explain what went wrong and what you are trying to do specifically. 

The mosaicjson seems to accept a json encoded colormap from the docs:
![image](https://github.com/user-attachments/assets/92b428ba-15f7-4135-a8ea-39ba0364ac60)

**Comment by vincentsarago** [comment]
you can also have a look at https://developmentseed.org/titiler/advanced/rendering/

**Reply by keepbeer** [reply]
hi@vincentsarago ,@zacharyDez, thanks for the response.

 I make a colormap.json file like this,
{
  "85": [155, 155, 155, 255],
  "179": [209, 209, 209, 255],
  "67": [65, 0, 0, 255],
  "53": [0, 61, 0, 255],
  "104": [0, 109, 0, 255],
  "152": [0, 152, 0, 255],
  "70": [0, 0, 91, 255],
  "183": [0, 0, 214, 255]
}
but it don't work, and  it prompt like this ---> "detail": "Could not parse the colormap value."

Request URL:
http://127.0.0.1:8000/mosaicjson/tiles/WorldMercatorWGS84Quad/9/413/203@1x.png?url=file%3AD%3A%5Ctitiler_proj%5Ctitiler-main%5Cmosaic.json&bidx=1&bidx=2&bidx=3&expression=b1%2Fb2&pixel_selection=mean&rescale=0%2C3000&colormap=file%3AD%3A%5Ctitiler_proj%5Ctitiler-main%5Ccolormap.json

**Reply by vincentsarago** [reply]
you can't pass the colormap using a file, you need to encode the JSON 

as shown in https://developmentseed.org/titiler/advanced/rendering/#custom-colormaps

**Reply by keepbeer** [reply]
thanks @vincentsarago for response,Is that only support the tiff of one band? If i want to change some color for the tiff with 3 bands just like the photoshop do, how can i support to do that?
